modules/server/src/main/scala/navigate/server/NavigateEngine.scala

```diff
@@ -410,4 +464,65 @@ object NavigateEngine {
     }
   )
 
+  private def command[F[_]: MonadThrow: Logger](
+    engine:  StateEngine[F, State, NavigateEvent],
+    cmdType: NavigateCommand,
+    cmd:     Handler[F, State, Event[F, State, NavigateEvent], Unit],
+    f:       Lens[State, Boolean]
```

Comment:
What does f zooms into?

Reply:
It zooms into the guard flag for the command in the global state. The execution of the command has three parts 1. The state machine part, which checks the flag and either sets it, or rejects the command. If the flag is not set, the command is added to the Streams of Streams.
2. The evaluation of the command. This is done in the join of streams to avoid blocking the state machine.
3. The result of the command is processed in the state machine. The guard flag is cleared, and the command result sent in the output stream.

Reply:
Could we put this as a comment on the code for future generations?